lib/rules/no-container.ts

```diff
@@ -0,0 +1,82 @@
+import { ESLintUtils, TSESTree } from '@typescript-eslint/experimental-utils';
+import { getDocsUrl } from '../utils';
+import {
+  isCallExpression,
+  isIdentifier,
+  isMemberExpression,
+  isObjectPattern,
+  isProperty,
+  isRenderVariableDeclarator,
+} from '../node-utils';
+
+export const RULE_NAME = 'no-container';
+
+export default ESLintUtils.RuleCreator(getDocsUrl)({
+  name: RULE_NAME,
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Disallow the use of container methods',
+      category: 'Best Practices',
+      recommended: 'error',
+    },
+    messages: {
+      noContainer:
+        'Unexpected use of container methods. Prefer the use of "screen.someMethod()".',
```

Comment:
What do you think about this message? Lifted from Testing Library docs.

```suggestion
        'Avoid using container to query for elements. Use some Testing Library query instead.',
```

Reply:
Though I just realized it could mean any method, not just queries.

Reply:
Yes, it's a good move to keep consistency between lint messages and the lib's docs. I think we could be more descriptive with the suggestion of what to do, so by mixing your suggested message with the one from @gndelia we could get something like:
`Avoid using container to query for elements. Prefer using query methods from Testing Library, such as "getByRole()"`, what do you think?

Reply:
Sounds good to me, but I'd wait for @Belco90  confirmation as well. Minor (my English is not native so I might be wrong), but I think it should be `Avoid using container to query elements` (dropping for)

Reply:
It didn't sound right to me but I just got it from Testing Library doc so I didn't say anything about rewording it.